x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/use_bulk_actions.tsx

```diff
@@ -276,14 +282,10 @@ export const useBulkActions = ({
           toasts,
           payload: { edit: [editPayload] },
           onFinish: () => hideWarningToast(),
-          search: isAllSelected
-            ? {
-                query: convertRulesFilterToKQL({
-                  ...filterOptions,
-                  showCustomRules: true, // only edit custom rules, as elastic rule are immutable
-                }),
-              }
-            : { ids: customSelectedRuleIds },
+          search: prepareSearchParams({
+            ...(isAllSelected ? { filterOptions } : { selectedRuleIds }),
+            dryRunResult: getBulkActionsDryRunFromCache(queryClient),
```

Comment:
Why are cached dry run results used here? Cached values are not reliable as they could be evicted and, in general, should not be used directly.

Reply:
It's the same as current implementation(only other cache key used):
https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/use_bulk_actions.tsx#L241-L243

I believe, we discussed it with you on when current implementation was done. It was an effort to move from imperative call within bulk method itself to declarative by using react-query: https://github.com/elastic/kibana/pull/122635#discussion_r793823176
So the idea was:
- trigger call from bulk edit method for parent hook, which retrieve data
- wait until it resolved
- use the result of it inside method 

To retrieve that results, we used function to get it from cache. In our case, just refreshed cache.
So, the value here is not stale, as was just updated